File: backend/src/main/java/com/woowacourse/friendogly/member/repository/MemberRepository.java
package com.woowacourse.friendogly.member.repository;

import com.woowacourse.friendogly.exception.FriendoglyException;
import com.woowacourse.friendogly.member.domain.Member;
import org.springframework.data.jpa.repository.JpaRepository;

public interface MemberRepository extends JpaRepository<Member, Long> {

    default Member getById(Long id) {
        return findById(id).orElseThrow(() -> new FriendoglyException("회원 정보를 찾지 못했습니다."));
    }
}

File: android/app/src/main/java/com/woowacourse/friendogly/domain/repository/MemberRepository.kt
package com.woowacourse.friendogly.domain.repository

import com.woowacourse.friendogly.domain.model.Member
import okhttp3.MultipartBody

interface MemberRepository {
    suspend fun postMember(
        name: String,
        email: String,
        file: MultipartBody.Part?,
    ): Result<Member>

    suspend fun getMemberMine(): Result<Member>
}
